app/models/concerns/arel_helper.rb

```diff
@@ -155,9 +155,10 @@ def self.one_for_column(column, source_arel_table:, group_on:, direction: :desc,
       end
 
       direction = :desc unless direction.in?([:asc, :desc])
+      group_columns = Array.wrap(group_on).map { |c| source_arel_table[c] }
 
-      max_by_group = source.distinct_on(source_arel_table[group_on]).
-        order(source_arel_table[group_on], source_arel_table[column].send(direction))
+      max_by_group = source.distinct_on(group_columns).
```

Comment:
Should this be `*group_columns` like the next line?

Reply:
You would think, but that threw an error about arguments.  It seemed happier with an array.